include/deal.II/base/mu_parser_internal.h

```diff
@@ -0,0 +1,100 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mu_parser_internal_h
+#define dealii_mu_parser_internal_h
+
+// This file contains functions used internally by the FunctionParser
+// and the TensorFunctionParser class.
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/thread_management.h>
+
+#include <boost/random.hpp>
+
+#include <cmath>
+#include <iterator>
+#include <map>
+#include <memory>
+#include <vector>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+
+#ifdef DEAL_II_WITH_MUPARSER
+
+namespace internal
+{
+  int
+  mu_round(double val);
+
+  double
+  mu_if(double condition, double thenvalue, double elsevalue);
+
+  double
+  mu_or(double left, double right);
+
+  double
+  mu_and(double left, double right);
+
+  double
+  mu_int(double value);
+
+  double
+  mu_ceil(double value);
+
+  double
+  mu_floor(double value);
+
+  double
+  mu_cot(double value);
+
+  double
+  mu_csc(double value);
+
+  double
+  mu_sec(double value);
+
+  double
+  mu_log(double value);
+
+  double
+  mu_pow(double a, double b);
+
+  double
+  mu_erfc(double value);
+
+  // returns a random value in the range [0,1] initializing the generator
+  // with the given seed
+  double
+  mu_rand_seed(double seed);
+
+  // returns a random value in the range [0,1]
+  double
+  mu_rand();
+
+  extern std::vector<std::string> function_names;
```

Comment:
The internal namespace is, in essence, a global namespace for functions and classes throughout deal.II. That's ok for the functions above, since they are all prefixed with `mu_` and so, in some sense, namespace themselves. But `function_names` is a different issue: It's a generic name without explanation which part of deal.II it belongs to. I think all of what you have here should actually be in a further nested namespace `internal::FunctionParser` to make things clearer.

Reply:
Done.